cpp/src/binaryop/compiled/binary_ops.cu

```diff
@@ -119,9 +119,9 @@ struct compare_functor {
 
   // This is used to compare a scalar and a column value
   template <typename LhsViewT = LhsDeviceViewT, typename RhsViewT = RhsDeviceViewT>
-  CUDA_DEVICE_CALLABLE typename std::enable_if_t<std::is_same_v<LhsViewT, column_device_view> &&
-                                                   !std::is_same_v<RhsViewT, column_device_view>,
-                                                 OutT>
+  CUDF_DI typename std::enable_if_t<std::is_same_v<LhsViewT, column_device_view> &&
```

Comment:
do we need every function to be `inline`?

Reply:
Probably not, but to be clear `CUDA_DEVICE_CALLABLE` _already_ implied inline. This PR simply renames the macro so that the behavior is clear. I considered introducing `CUDF_HD` and `CUDF_D` in this PR to replace `__host__ __device__` without inline, but I decided to keep this PR a narrow renaming only. My hope is that the rename will force people to ask the question that you're asking, though, because I think a lot of places are using `CUDA_HOST_DEVICE_CALLABLE` currently that don't need to be inlined.

Reply:
Makes sense. 
Might be good to add `CUDF_HD` in this PR, even if unused for now. Less friction to make the right choice going forward.

Reply:
Sounds good, I was debating whether that would be a good idea or not. I'll add it.